packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx

```diff
@@ -28,12 +79,150 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
   );
 
   return (
-    <AFFiNEDatePicker
-      inline
-      value={date}
-      onSelect={onDateSelect}
-      calendarClassName={styles.calendar}
-    />
+    <div className={styles.journalPanel}>
+      <AFFiNEDatePicker
+        inline
+        value={date}
+        onChange={onDateSelect}
+        calendarClassName={styles.calendar}
+      />
+
+      <JournalDailyCountBlock {...props} />
+    </div>
+  );
+};
+
+const sortPagesByDate = (
+  pages: Page[],
+  field: 'updatedDate' | 'createDate',
+  order: 'asc' | 'desc' = 'desc'
+) => {
+  return [...pages].sort((a, b) => {
+    return (
+      (order === 'asc' ? 1 : -1) *
+      dayjs(b.meta[field]).diff(dayjs(a.meta[field]))
+    );
+  });
+};
+
+const DailyCountEmptyFallback = ({ name }: { name: NavItemName }) => {
+  const t = useAFFiNEI18N();
+
+  return (
+    <div className={styles.dailyCountEmpty}>
+      {name === 'createdToday'
+        ? t['com.affine.journal.daily-count-created-empty-tips']()
+        : t['com.affine.journal.daily-count-updated-empty-tips']()}
+    </div>
+  );
+};
+const JournalDailyCountBlock = ({ workspace, page }: EditorExtensionProps) => {
+  const nodeRef = useRef<HTMLDivElement>(null);
+  const t = useAFFiNEI18N();
+  const { journalDate } = useJournalInfoHelper(page?.meta);
+  const [createdToday, setCreatedToday] = useState<Page[]>([]);
+  const [updatedToday, setUpdatedToday] = useState<Page[]>([]);
+  const [activeItem, setActiveItem] = useState<NavItemName>('createdToday');
+
+  const navigateHelper = useNavigateHelper();
+
+  const headerItems = useMemo<NavItem[]>(
+    () => [
+      {
+        name: 'createdToday',
+        label: t['com.affine.journal.created-today'](),
+        count: createdToday.length,
+      },
+      {
+        name: 'updatedToday',
+        label: t['com.affine.journal.updated-today'](),
+        count: updatedToday.length,
+      },
+    ],
+    [createdToday.length, t, updatedToday.length]
+  );
+
+  useEffect(() => {
+    const created: Page[] = [];
+    const updated: Page[] = [];
+    const today = journalDate;
+    Array.from(workspace.pages.values()).forEach(page => {
+      if (page.meta.trash) return;
+      if (
+        page.meta.createDate &&
+        dayjs(page.meta.createDate).isSame(today, 'day')
+      ) {
+        created.push(page);
+      }
+      if (
+        page.meta.updatedDate &&
+        dayjs(page.meta.updatedDate).isSame(today, 'day')
+      ) {
+        updated.push(page);
+      }
+    });
+    setCreatedToday(sortPagesByDate(created, 'createDate'));
+    setUpdatedToday(sortPagesByDate(updated, 'updatedDate'));
```

Comment:
Seems the created list and updated list should be calculation memos, not a state.